mesonbuild/coredata.py

```diff
@@ -158,6 +158,12 @@ def validate_value(self, value, user_input=True):
                     ', '.join(bad), ', '.join(self.choices)))
         return newvalue
 
+class UserFeatureOption(UserComboOption):
+    static_choices = ['enabled', 'disabled', 'automatic']
```

Comment:
I also like `true`/`false`/`auto` or `yes`/`no`/`auto`. We will have to / should support `true`/`false` in any case, as people will turn boolean options into feature options with autodetect and that should be smooth. Also it's the most intuitive from a user point of view I think.

Reply:
Actually, thinking again about this, true/false is confusing because `false` would have 2 different meanings when passed to `required' kwarg.

- `required: false` has the behaviour of 'auto'.
- `required: get_option('an-option-set-to-false')` has the behaviour of a disabled feature.

I'm changing my mind in favour of `enabled/disabled/auto` because that's the more explicit.

> The return value of get_option() should be an opaque object with the following methods on it: enabled(), disabled(), auto().

See, why would you have a method `enabled()` if the value is `true`? 

We say in English: "You should **enable** this feature" and not "You should **true** this feature" or "You should **yes** this feature".